Cleaned and tidied code. Added basic firestore service support.

emrothenberg opened the pull request:
- Added Firestore support: Introduced basic functionality for interacting with Firestore, enabling read and update operations through the Pyrebase4 library, with option to add create and delete for complete CRUD support.
- Code cleanup and refactoring: Improved code readability and maintainability by cleaning and tidying up existing code.

nhorvath commented:
What were the material changes to the files that show completely changed? I prefer commits that use a linter for the first time to have no code changes so they can be ignored.

emrothenberg commented:
1. Created a "services" module and moved all the services there (auth, database, storage),
2. Moved non-service classes to new module "types"
3. Created "utils" module to populate repeatedly used functions, since now not all elements are in the same file
4. Added a new service to include Firestore in `services/firestore.py`. This is the only functional change.
5. All changes after the first commit were only done to the firestore file, as well as updating the README to document functionality.

nhorvath commented:
Just curious, why did you close this PR?
